access/types/forms.py

```diff
@@ -146,7 +153,7 @@ def __init__(self, *args, **kwargs):
                     fi.group_errors = group_errors
 
                 if j == 0:
-                    f[0].open_set = self.group_name(g)
+                    f[0].open_set = "{}_{}".format(exercise_key, self.group_name(g))
```

Comment:
I guess this could be `"exercise-{}-set-{}".format(self.form_nonce, self.group_name(g))`